src/components/Calculator.js

```diff
@@ -1,86 +1,283 @@
+    this.addNumber.bind = this.addNumber.bind(this);
   }
 
+  proccessResult = (result) => {
+    this.setState(() => ({
+      display: result.total,
+      total: result.total,
+      next: result.next,
+      operation: result.operation,
+      toNumber1: true,
+    }));
+  };
+
+  selectOperator = (event) => {
+    event.preventDefault();
+    const option = event.target.innerText;
+    if (
+      option === '+/-'
+      || option === '%'
+      || option === '÷'
+      || option === 'x'
+      || option === '-'
+      || option === '+'
+    ) {
+      this.setState(() => ({
+        operation: option,
+        toNumber1: false,
+      }));
+    }
+
+    if (option === 'AC') {
+      const { total, next, operation } = this.state;
+      const obj = { total, next, operation };
+      const result = calculate(obj, option);
+      this.proccessResult(result);
+    }
+
+    if (option === '.') {
+      const { total, next, operation } = this.state;
+      const obj = { total, next, operation };
+      const result = calculate(obj, option);
+      this.setState(() => ({
+        total: result.total,
+        display: result.total,
+      }));
+    }
+
+    if (option === '=') {
+      const { total, next, operation } = this.state;
+      const obj = { total, next, operation };
+      const result = calculate(obj, option);
+      this.proccessResult(result);
+    }
+  };
+
+  addNumber = (event) => {
+    event.preventDefault();
+    const {
+      toNumber1, total, operation, next, display,
+    } = this.state;
+    let newTotal = total;
+    const number = event.target.innerText;
+    let numberResult = null;
+    let float = null;
+    if (total) {
+      float = total.indexOf('.');
+    }
+
+    if (display !== '0' && !next && !operation && !float) {
+      newTotal = null;
+    }
+
+    if (toNumber1 === true) {
+      numberResult = newTotal;
+      if (numberResult) {
+        numberResult += number;
+      } else {
+        numberResult = number;
+      }
+
+      this.setState(() => ({
+        total: numberResult,
+        display: numberResult,
+      }));
+    } else {
+      numberResult = next;
+      if (numberResult) {
+        numberResult += number;
+      } else {
+        numberResult = number;
+      }
+      const display = `${newTotal} ${operation} ${numberResult}`;
+      this.setState(() => ({
+        next: numberResult,
+        display,
+      }));
+    }
+  };
+
   render() {
-    const { result } = this.state;
+    const { display } = this.state;
 
     return (
       <div className="calculator-body">
         <div className="body-result">
-          <h5>{result}</h5>
+          <h5>{display}</h5>
         </div>
         <div className="body-buttons">
           <div className="buttons-row-1">
-            <button type="button" className="col-1 ">
+            <button
+              onClick={this.selectOperator}
+              name="AC"
+              type="button"
+              className="col-1 "
+            >
```

Comment:
- [ ] OPTIONAL: The Calculator component can be restructured to smaller child components. For example `Button`, `ButtonPanel` and `Display` can construct the `Calculator` component. You did a great job, though. :+1: